src/dice/ffi/metall_internal.hpp

```diff
@@ -5,11 +5,42 @@
 #include <metall/metall.hpp>
 
 namespace dice::metall_ffi::internal {
-    /**
+#if METALL_VERSION >= 2800
```

Comment:
with what versions <=0.28 have you tested it?

Reply:
the `read_only()` function was added in 0.28 thats the only reason why this exists

Reply:
tested with 0.23.1

Reply:
For safety, I would suggest to add
```
#if !defined(METALL_VERSION) || METALL_VERSION < 2301
#error "metall must be at least v0.23.1 (METALL_VERSION 2301)"
#endif
```

Reply:
why? there is nothing that can be incompatible here, its just a struct with an additional bool

Reply:
The point of the distinction here is literally just: does this have a `read_only()` function

Reply:
I was more thinking about all the other interfaces. But on second thought, they should all come from boost. So, never mind.